src/sonic-yang-models/yang-models/sonic-sflow.yang

```diff
@@ -32,10 +32,28 @@ module sonic-sflow{
 
     description "SFLOW yang Module for SONiC OS";
 
+    revision 2023-04-11 {
+        description "Add direction command to support egress sflow";
+    }
+
     revision 2021-04-26 {
         description "First Revision";
     }
 
+    typedef sample_direction {
```

Comment:
As per https://github.com/openconfig/public/blob/master/release/models/sampling/openconfig-sampling-sflow.yang it assumes that the interface will perform both ingress and egress sampling without the need of direction leaf? If that is the case when we support latest https://github.com/openconfig/public/blob/master/release/models/sampling/openconfig-sampling-sflow.yang in community sonic we should always populate direction as "both" in sonic-flow.yang?

Reply:
As mentioned in above comment, as per sflow spec, an interface must have same sample-rate on ingress and egress. Since initial support in SONiC was handling only Ingress, default value is set to "rx" for backward compatibility. Pls check HLD for more info.